posydon/visualization/plot_pop.py

```diff
@@ -251,4 +258,137 @@ def plot_hist_properties(x, df_intrinsic=None, df_observable=None,
     if path:
         plt.savefig(path)
     if show:
-        plt.show()
\ No newline at end of file
+        plt.show()
+        
+        
+def plot_popsyn_over_grid_slice(pop, grid_type, met_Zsun, slices=None, channel=None, 
+                                plot_dir='./', prop=None, prop_range=None, 
+                                log_prop=False, alpha=0.3, s=5., 
+                                show_fig=True, save_fig=True, close_fig=True,
+                                verbose=False):
+
+    # load grid
+    met = convert_metallicity_to_string(met_Zsun)
+    grid_path = os.path.join(PATH_TO_POSYDON_DATA, f'POSYDON_data/{grid_type}/{met}_Zsun.h5')
+    grid = PSyGrid(verbose=False)
+    grid.load(grid_path)
+
+    # check if formation channel information is avaialbe
+    if channel is not None and 'channel' not in pop.df_oneline.keys():
+        if verbose:
+            print('Computing formation channels...')
+        pop.get_formation_channels()
+        
+    if 'CO' in grid_path:
+        # compact object mass slices
+        def log_range(x_min,x_max,x_n):
+            return 10**np.linspace(np.log10(x_min),np.log10(x_max), x_n)
+        m_CO_min = 1.
+        m_CO_max = 51.32759630397827
+        m_CO_n = 23
+        m_COs = log_range(m_CO_min, m_CO_max, m_CO_n)
+        m_COs_edges = 10**(np.log10(np.array(m_COs)[:-1])+
+                                (np.log10(np.array(m_COs)[1:])-np.log10(np.array(m_COs)[:-1]))/2)
+        m2 = [0.]+m_COs_edges.tolist()+[55.]
```

Comment:
Again, don't hard code values as long as they aren't fixed limits.
So, I'd either expect:
- in the case of having limits: `m2 = [0.]+m_COs_edges.tolist()+[np.finfo(a.dtype).max]`
- in the case of expanding the range: `m2 = [10**(2*np.log10(m_COs_edges[0])-np.log10(m_COs_edges[1]))]+m_COs_edges.tolist()+[10**(2*np.log10(m_COs_edges[-1])-np.log10(m_COs_edges[-2]))]` here it would be good to add an catch in case the array don't have 1 or 0 values.

Reply:
fixed